### PR TITLE
Disable warnings-as-errors for release branches

### LINKED ISSUE #33627: Disable warnings-as-errors in release branches
The noisy warnings-as-errors are ongoing problem in release branches. See the discussion https://github.com/dotnet/coreclr/pull/28026#issuecomment-598952586

### PR DESCRIPTION
**Pull request by jkotas**
Fixes #33627

### CODE REVIEW

**Review by safern — COMMENTED** [review]
Does this mean that warnings as errors will be disabled in master as well? I see the PR targets master and PRERELEASE is set to `1`.

**Comment by jkotas** [comment]
The warnings as errors are still going to be enabled in master. 

This is a prep-work for when we flip PRERELEASE to 0 in release branches - it is what is going to disable warnings as errors.

**Comment by safern** [comment]
Ah right, I miss read the condition.

**Review by safern — APPROVED** [review]

File: src/coreclr/tests/src/Interop/PInvoke/Generics/CMakeLists.txt [hunk 1]
```diff
@@ -13,6 +13,10 @@ elseif(CLR_CMAKE_TARGET_ARCH_ARM64)
   add_definitions(-DTARGET_ARM64)
   add_definitions(-DTARGET_ARMARCH)
 endif()
+if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+  # The ABI for passing parameters with 32-byte alignment has changed in GCC 4.6
+  add_compile_options(-Wno-psabi)
```
Comment (am11) [inline src/coreclr/tests/src/Interop/PInvoke/Generics/CMakeLists.txt]: @jkotas, just wondering if warning as error state for PR runs is same as before (`ON`), why was this special case required for GCC as master was ok without it?

Comment (jkotas) [reply src/coreclr/tests/src/Interop/PInvoke/Generics/CMakeLists.txt]: The PR enabled warnings-as-errors for tests. The CoreCLR tests were built without `-Werror` before this change.

Comment (am11) [reply src/coreclr/tests/src/Interop/PInvoke/Generics/CMakeLists.txt]: Thanks, makes sense.

**Pull request merged**